notebooks/02_model/rbm_deep_dive.ipynb

```diff
@@ -0,0 +1,1154 @@
+    "\n",
+    "![rbm1](https://recodatasets.blob.core.windows.net/images/RBM1.png)\n",
+    "\n",
+    "The input of the movielens database consists of ratings from 1 to 5; we shall thus consider a discrete configuration space of $m$ visible variables, each taking values in a finite set $\\chi_v = \\{ 1, 2, 3,4,5 \\}$. A global configuration of the system is determined by $\\mathbf{v} = (v_1, v_2, ..., v_m) \\in \\chi_v^m$ and we reserve $0$ for an unrated movie. We also need to specify the hidden units, that we take as random binary variables $\\chi_h = \\{0,1 \\}$ denoting if the particular unit is active or not and $\\mathbf{h} = (h_1, h_2, ...,h_n) \\in \\chi_h^n$. The hidden units may describe attributes such as the genre of a movie; for example, given a sci-fi/horror movie, only the hidden units describing such attributes should be active. The minimal model for such a system is defined by the following Hamiltonian: \n",
+    "\n",
+    "$$H = - \\sum_{i,j \\in G} v_i \\, w_{ij} \\, h_j - \\sum_{i=1}^m v_i \\, a_i - \\sum_{j=1}^n h_i \\, b_i$$\n",
+    "\n",
+    "The first term is an \"interaction term\", capturing the correlations between the visible and hidden units, while the other two terms are \"potential terms\", taking into account the bias of the units. The correlation matrix $w_{ij}$ and the two biases $a_i$ and $b_i$ are learning parameters to be fixed by the minimization of a properly defined cost function. Remember that this is an unsupervised problem, i.e. there is no real output and therefore we cannot directly minimize the error function between the prediction and the labeled data. As in every SM problem, the right quantity to minimize is the Free energy (remember that $\\beta =1$)\n",
+    "\n",
+    "$$ F =- \\log Z =- \\log \\sum_{ v_i, h_i } P(v, h) $$.\n",
+    "\n",
+    "In the language of probability theory, the above quantity is the cumulant generating function. One way of evaluating the free energy is to use a [Markov-chain Montecarlo sampling](https://en.wikipedia.org/wiki/Monte_Carlo_method#Computer_graphics) algorithm such as the Metropolis-Hasting; here we will use instead an approximate method called Contrastive divergence, based on [Gibbs sampling](https://en.wikipedia.org/wiki/Gibbs_sampling) (see below). The latter has the advantage of being faster than Montecarlo. Once the candidate $F$ has been found, we fix the learning parameters by extremizing $F$. Let us see how this works in practice in the next section. "
```

Comment:
extremizing -> minimizing

Reply:
Thank I will change it to minimising. What I meant is that we are only evaluating the first order derivative, so we are only searching for optima; only looking at the Hessian we know the nature of the optima. In most of the cases in ML there is no guarantee we reach the minimum as the problem is non-convex and most likely we end up in some low error saddle point. But yes, at the end our intention is to minimise the cost function, so the second term is more appropriate in this context.